src/realm/table.cpp

```diff
@@ -770,8 +770,16 @@ void Table::populate_search_index(ColKey col_key)
             }
         }
         else if (type == type_String) {
-            StringData value = o.get<StringData>(col_key);
-            index->insert(key, value); // Throws
+            if (col_key.is_list()) {
+                auto list = o.get_list<String>(col_key);
+                for (auto& s : list) {
+                    index->insert(key, s); // Throws
```

Comment:
This will insert duplicate values if they exist in the list. Elsewhere, that isn't allowed, shouldn't this protect against that as well?

Reply:
Good catch. I made this first - before really thinking of duplicates.